devtools/server/actors/replay/auth.js

```diff
@@ -224,32 +242,29 @@ async function refresh() {
     const json = await resp.json();
 
     if (json.error) {
-      pingTelemetry("browser", "auth-request-failed", {
-        message: json.error,
-        authId: lastAuthId
-      });
-      setReplayRefreshToken("");
-      setReplayUserToken("");
-      return;
+      throw new Error(json.error);
     }
 
-    if (json.access_token) {
-      Services.prefs.setStringPref("devtools.recordreplay.refresh-token", json.refresh_token);
-      setReplayUserToken(json.access_token);
-
-      // refresh a minute before token expiration
-      setTimeout(refresh, json.expires_in * 1000 - (60 * 1000));
-    } else {
-      pingTelemetry("browser", "auth-request-failed", {
-        message: "no-access-token",
-        authId: lastAuthId
-      });
+    if (!json.access_token) {
+      throw new Error("no-access-token");
     }
+
+    Services.prefs.setStringPref("devtools.recordreplay.refresh-token", json.refresh_token);
+    setReplayUserToken(json.access_token);
+
+    scheduleRefreshTimer(json.expires_in * 1000);
+
+    return json.access_token;
   } catch (e) {
     pingTelemetry("browser", "auth-request-failed", {
-      message: String(e),
+      message: e.message ? e.message : String(e),
       authId: lastAuthId
     });
+
+    setReplayRefreshToken("");
```

Comment:
We were previously only clearing the tokens on error from Auth0. This version will clean up tokens on any error.